pkg/config/remote_config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// supported remote config provider schemes
+const (
+	HTTP  = "http"
+	HTTPS = "https"
+	// TODO: add s3, gcs, blob, and git providers backed by go-fsimpl
+)
+
+// RemoteOpts struct contains agent remote config options
+type RemoteOpts struct {
+	URL              *url.URL
+	HTTPClientConfig *config.HTTPClientConfig
+}
+
+// RemoteProvider ...
+type RemoteProvider interface {
+	Retrieve() ([]byte, error)
+}
+
+// NewRemoteConfig ...
+func NewRemoteConfig(rawURL string, opts *RemoteOpts) (RemoteProvider, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+	if opts == nil {
+		// Default provider opts
+		opts = &RemoteOpts{}
+	}
+	opts.URL = u
+
+	switch u.Scheme {
+	case "":
+		// if no scheme, assume local file path, return nil and let caller handle.
+		return nil, nil
+	case HTTP, HTTPS:
+		httpP, err := newHTTPProvider(opts)
+		if err != nil {
+			return nil, err
+		}
+		return httpP, nil
+	default:
+		return nil, fmt.Errorf("remote config scheme not supported: %s", u.Scheme)
+	}
+}
+
+// Remote Config Providers
+// httpProvider - http/https provider
+type httpProvider struct {
+	myURL            url.URL
+	httpClientConfig config.HTTPClientConfig
```

Comment:
nit: I'd just store the opts here
(second nit: keep myURL as a pointer, since that seems to be common)

Reply:
thinking about this, I probably only need to store the `http.Client` here instead of the opts.